app/org/sagebionetworks/bridge/upload/UploadValidationContext.java

```diff
@@ -100,57 +100,66 @@ public void addMessage(String msg) {
         messageList.add(msg);
     }
 
-    /** Raw upload data as bytes. This is created by S3DownloadHandler and read by the DecryptHandler. */
-    public byte[] getData() {
-        return data;
+    /** Temporary directory in which we process the upload. */
+    public File getTempDir() {
+        return tempDir;
     }
 
-    /** @see #getData */
-    public void setData(byte[] data) {
-        this.data = data;
+    /** @see #getTempDir */
+    public void setTempDir(File tempDir) {
+        this.tempDir = tempDir;
     }
 
-    /** Decrypted upload data as bytes. This is created by DecryptHandler and read by UnzipHandler. */
-    public byte[] getDecryptedData() {
-        return decryptedData;
+    /** Raw upload data file. This is created by S3DownloadHandler and read by the DecryptHandler. */
+    public File getDataFile() {
+        return dataFile;
     }
 
-    /** @see #getDecryptedData */
-    public void setDecryptedData(byte[] decryptedData) {
-        this.decryptedData = decryptedData;
+    /** @see #getDataFile */
+    public void setDataFile(File dataFile) {
+        this.dataFile = dataFile;
+    }
+
+    /** Decrypted upload data file. This is created by DecryptHandler and read by UnzipHandler. */
+    public File getDecryptedDataFile() {
+        return decryptedDataFile;
+    }
+
+    /** @see #getDecryptedDataFile */
+    public void setDecryptedDataFile(File decryptedDataFile) {
+        this.decryptedDataFile = decryptedDataFile;
     }
 
     /**
-     * Unzipped data as bytes, keyed by filename. This is initially created by the UnzipHandler. The ParseJsonHandler
-     * will read this and remove entries that can be parsed into JSON. Non-JSON entries will still remain in this map.
-     * This is also read by the IosSchemaValidationHandler.
+     * Unzipped data files, keyed by filename. This is created by the UnzipHandler and read by InitRecordHandler, as
+     * well as UploadFormatHandler, and its inner handlers.
      */
-    public Map<String, byte[]> getUnzippedDataMap() {
-        return unzippedDataMap;
+    public Map<String, File> getUnzippedDataFileMap() {
+        return unzippedDataFileMap;
     }
 
-    /** @see #getUnzippedDataMap */
-    public void setUnzippedDataMap(Map<String, byte[]> unzippedDataMap) {
-        this.unzippedDataMap = unzippedDataMap;
+    /** @see #getUnzippedDataFileMap */
+    public void setUnzippedDataFileMap(Map<String, File> unzippedDataFileMap) {
+        this.unzippedDataFileMap = unzippedDataFileMap;
     }
```

Comment:
What are the error conditions for writing to file? Can the drive fill up? Do we want to write to an EFS "drive"? Do we know what the hard limit is on filesystem storage is on our instances?

Reply:
We can configure an EBS drive in Elastic Beanstalk and/or EC2. It's currently set to "container default", and for Exporter, that looks to be about 8gb, 75% of which is free.

We also delete temp dirs when we're done using them, and the temp dirs are created in _the_ temp dir, so I don't anticipate running out of disk.